### DISCUSSION TITLE
What is the use of "data" object returned by uploading file to supabase storage? I see nowhere it is used.

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#4771] [Questions] [answered]
data object contains `key` which contains path of file including bucket name also and i see no use of it. if it does not contain bucket name then it can be used to retrieve publicURL. or to download. 
```
const { data, error } = await supabase
  .storage
  .from('avatars')
  .upload('public/avatar1.png', avatarFile, {
    cacheControl: '3600',
    upsert: false
  })
```
here data.key contains `avatars/public/avatar1.png` so what i will do with this key? because later on for downloading or for getting publicURL i need `public/avatar1.png` and not `avatars/public/avatar1.png`

**Comment by inian** [comment, marked as answer]
Hi @roker15, upload will not return the bucket name in the next major release of storage-js https://github.com/supabase/storage-js/issues/6

**Reply by roker15** [reply]
when you are realeasing? also what is v1 in following public url? Is it version of storage? will this url break in future if storage vsersion is upgraded?
https://hbvffqslxssdbkdxfqop.supabase.co/storage/v1/object/public/notes-images/3463ee30-19f4-4b05-951c-b5216490ff7e-CodeCogsEqn.gif

**Reply by inian** [reply]
We have not decided on the launch date for the next major version yet. The URL will remain the same and won't break. The data returned from the client library will not have the bucket name anymore and when you upgrade your client library to the next major version, you will need to update your application code accordingly.
